docs/spec/v1beta1/gitrepositories.md

```diff
@@ -170,6 +177,42 @@ spec:
 
 When specified, `spec.ignore` overrides the default exclusion list.
 
+## Git Implementation
+
+You can skip this section unless you know that you need support for either
+specific git wire protocol functionality. Changing the git implementation
+comes with its own set of drawbacks.
+
+Some git providers like Azure DevOps require that the git client supports specific capabilities
+to be able to communicate. The initial library used in source-controller did not support
+this functionality while other libraries that did were missinging other critical functionality,
+specifically the ability to do shallow cloning. Shallow cloning is important as it allows
+source-controller to only fetch the latest commits, instead of the whole git history.
+For some very large repositories this means downloading GB of data that could fill the disk
+and also impact the traffic costs.
+
+To be able to support Azure DevOps a compromise solution was built, giving the user the
+option to select the git library while accepting the drawbacks.
+
+| Git Implementation | Shallow Clones | Azure DevOps Support |
+|---|---|---|
+| 'go-git' | true | false |
+| 'git2go' | false | true |
```

Comment:
I would name the v2 protocol support here, instead of just calling it "Azure DevOps Support", as that makes it easier to find whenever someone else runs into this issue on a platform that is not Azure DevOps.